src/main/ruby/truffleruby/core/exception.rb

```diff
@@ -469,6 +469,28 @@ def initialize(sig, message = undefined)
     @signo = signo
     super(name_with_prefix)
   end
+
+  private
+
+  def reached_top_level
+    banned = ['VTALRM']
+    if !TruffleRuby.native?
+      banned.concat(['FPE', 'ILL', 'SEGV', 'USR1', 'QUIT'])
+    end
+    banned.map! { |name| Signal::Names[name] }
+    if banned.include?(@signo)
+      warn "not acting on top level SignalException for SIG#{Signal::Numbers[@signo]} as it is VM reserved"
```

Comment:
FWIW, MRI seems to just raise for SIGVTALRM too.

Reply:
But, it seems it doesn't work if we try to raise(3) for SIGVTALRM, so I kept that behavior for SIGVTALRM, and raise for other signals like MRI.